haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -694,17 +710,25 @@ declsFromHsBinds ctx = dedupByTick . concatMap go
 isInstanceMethodVar :: Var -> Bool
 isInstanceMethodVar v = specialVar v == Just TypeclassySV
 
+-- | These are special bindings related to pattern synonyms which are not
+-- needed for us and contain overlapping declarations.
+isPatternSynonymSpecialVar :: Var -> Bool
+isPatternSynonymSpecialVar v = specialVar v == Just PatternSynonymSV
+
 specialVar :: Var -> Maybe SpecialVar
 specialVar v = case take 2 (nameOccurenceString (varName v)) of
     -- List appended as cases are encountered.
     "$d" -> Just $! TypeclassySV
```

Comment:
I suspect `$!` is doing nothing here, but I'd love to learn if I'm wrong.

Reply:
See https://johantibell.com/files/haskell-performance-patterns.html#(12) - and slide 10 as well for `return` generally. I'm also happy to learn the truth.